### ISSUE TITLE
Apply use of landmark roles

### ISSUE THREAD

**Issue by allinox** [Altinn/altinn-access-management-frontend#788]
![image](https://github.com/Altinn/altinn-access-management-frontend/assets/22012997/48394f2e-2615-4a20-b082-e26a5280b57d)
WCAG rule 1.3.1 requires each section of each page to be semantically described in the code, using (among other things) landmarks: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles#3._landmark_roles

Other than headers, we hardly use landmarks, and we should use them more.

The available landmarks are:
- [banner](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/banner_role) (document [`header`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header))
-  [complementary](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/complementary_role) ([`aside`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/aside))
-  [contentinfo](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/contentinfo_role) (document [`footer`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/footer))
-  [form](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/form_role) ([`form`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form))
-  [main](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/main_role) ([`main`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main))
-  [navigation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/navigation_role) ([`nav`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav))
-  [region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/region_role) ([`section`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section))
-  [search](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/search_role) ([`search`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/search))

And these are the ones we should be using. (Note: The goal is to make the structure of the page clear and navigation easier for users of assistive technologies. They should therefor not be overused since that would clutter the page, but should be chosen and used in a way that makes navigating the page as intuitive as possible and represents the functionality of the page)

For example should the role [`search`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/search_role) be assigned to the entire region of searches across our pages (search field, filter, search results). However, the chosen resources (the rightmost area in API-deleg and the CollectionBar area) should not be part of the search landmark.
Each page should also have a `main` that encompasses the main content of the page (such as the entire page in ApiDeleg and SingleRights)
[According to MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/region_role), the role "region" should be used very sparingly, only if there is a section that truly needs a landmark, but no other roles apply. There should rarely be more than one region on any page, if any at all.

Some of the places that need added landmarks: (There, might be more, and these need to be identified as part of this task)

- [ ] Main area of each page (might be there already, not sure)
- [ ] Search area of each page that contains search functionality
- [ ] Header area
- [ ] Area of chosen action bars/ list of chosen resources / receipt? (Unsure if these should have a region landmark or no landmark at all)
